library/utils/route-helpers/routeHelpers.ts

```diff
@@ -37,9 +37,9 @@ export const mapChildRoutes = (routes: RouteObject[]): RouteObject[] => {
 // matches .page $transition-duration in transitions.scss
 export const innerTransitionTimeout = 300;
 
-export const getInfraPath = (
-  route: InfraRoute,
+export const getInfraPath = <Path extends string>(
+  route: Path,
```

Comment:
If we set this as `InfraRoute` the the `params` must always define all possible parameters, resulting in errors like:
```
Argument of type '{ parentRegionId: string; regionId: string; }' is not assignable to parameter of type '{ parentRegionId: string | null; siteId: string | null; regionId: string | null; clusterName: string | null; id: string | null; }'.
  Type '{ parentRegionId: string; regionId: string; }' is missing the following properties from type '{ parentRegionId: string | null; siteId: string | null; regionId: string | null; clusterName: string | null; id: string | null; }': siteId, clusterName, idts(2345)
```
with this the `params` type is dynamically inferred from the `route` parameter